src/p11_child/p11_child_openssl.c

```diff
@@ -205,6 +206,72 @@ static const EVP_MD *get_dgst(CK_MECHANISM_TYPE ocsp_dgst)
     return dgst;
 }
 
+static char *get_issuer_subject_str(TALLOC_CTX *mem_ctx, X509 *cert)
+{
+    X509_NAME *issuer_name;
+    X509_NAME *subject_name;
+    char *tmp_str = NULL;
+    BIO *bio_mem = NULL;
+    int ret;
+    char *str = NULL;
+
+    issuer_name = X509_get_issuer_name(cert);
+    subject_name = X509_get_subject_name(cert);
+
+    if (issuer_name == NULL || subject_name == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing issuer or subject.\n");
+        return NULL;
+    }
+
+    bio_mem = BIO_new(BIO_s_mem());
+    if (bio_mem == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "BIO_new failed.\n");
+        return NULL;
+    }
+
+    ret = BIO_printf(bio_mem, "Issuer: [");
+    if (ret == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "BIO_printf failed.\n");
+        goto done;
+    }
+
+    ret = X509_NAME_print_ex(bio_mem, issuer_name, 0, XN_FLAG_ONELINE);
+    if (ret == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "X509_NAME_print_ex failed.\n");
+        goto done;
+    }
+
+    ret = BIO_printf(bio_mem, "] Subject: [");
+    if (ret == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "BIO_printf failed.\n");
+        goto done;
+    }
+
+    ret = X509_NAME_print_ex(bio_mem, subject_name, 0, XN_FLAG_ONELINE);
+    if (ret == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "X509_NAME_print_ex failed.\n");
+        goto done;
+    }
+
+    ret = BIO_printf(bio_mem, "]");
+    if (ret == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "BIO_printf failed.\n");
+        goto done;
+    }
+
+    if (BIO_get_mem_data(bio_mem, &tmp_str) == 0) {
+        DEBUG(SSSDBG_OP_FAILURE, "BIO_get_mem_data failed.\n");
+        goto done;
+    }
+
+    str = talloc_asprintf(mem_ctx, "%s", tmp_str);
```

Comment:
Nitpick: probably it makes sense to log an error in case of `talloc_asprintf` fail...

Reply:
done